src/coreclr/System.Private.CoreLib/src/System/Diagnostics/Debugger.cs

```diff
@@ -74,8 +75,10 @@ public static void NotifyOfCrossThreadDependency()
         // Posts a message for the attached debugger.  If there is no
         // debugger attached, has no effect.  The debugger may or may not
         // report the message depending on its settings.
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern void Log(int level, string? category, string? message);
+        public static void Log(int level, string? category, string? message) => LogInternal(level, category, message);
+
+        [DllImport(RuntimeHelpers.QCall, CharSet = CharSet.Unicode)]
+        private static extern void LogInternal(int level, string? category, string? message);
```

Comment:
For my own edification, why was this helper needed?  Can public methods not be QCalls?  Or it somehow bleeds into the public contract in a way we don't want it to?

Reply:
You will get build errors from the libraries tooling if you expose DllImport method directly in public surface.

I guess this restriction can be removed if we cared enough and fixed all places that depend on it currently.

Reply:
Presumably that's just https://github.com/dotnet/runtime/blob/e4c6c4c4422b15c526642ee139e89bda4f83b242/eng/CodeAnalysis.ruleset#L66 ?

I don't actually care "enough" :), just trying to understand.

What do you mean by "fixing all places that depend on it"?  Is there something in the runtime that doesn't like such methods?

Reply:
- Runtime asserts on public QCalls currently (https://github.com/dotnet/runtime/blob/master/src/coreclr/vm/ecall.cpp#L686). The comment in front of this assert suggest that it is no longer relevant and it should be safe to delete the assert. 
- GenAPI tooling may need some fixes
- Version resilient R2R may need some fixes

Reply:
Got it. Thanks.